packages/overlay/src/integrations/sentry/components/insights/InsightsTabDetails.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import { Navigate, Outlet, Route, Routes } from 'react-router-dom';
+import Tabs from '~/components/Tabs';
+import { useSpotlightContext } from '~/lib/useSpotlightContext';
+import { useSentrySpans } from '../../data/useSentrySpans';
+import HiddenItemsButton from '../HiddenItemsButton';
+import Queries from './Queries';
+import QuerySummary from './QuerySummary';
+import Resources from './Resources';
+import WebVitals from './webVitals';
+import WebVitalsDetail from './webVitals/WebVitalsDetail';
+
+export default function InsightsTabDetails() {
+  const context = useSpotlightContext();
+  const { allSpans, localSpans } = useSentrySpans();
+
+  const [activeTab, setActiveTab] = useState('');
+  const [showAll, setShowAll] = useState(!context.experiments['sentry:focus-local-events']);
+
+  const hiddenItemCount = allSpans.length - localSpans.length;
+
+  const tabs = [
+    {
+      id: 'queries',
+      title: 'Queries',
+      active: activeTab === 'queries',
+      onSelect: () => setActiveTab('queries'),
+    },
+    {
+      id: 'webvitals',
+      title: 'Web Vitals',
+      active: activeTab === 'webvitals',
+      onSelect: () => setActiveTab('webvitals'),
+    },
+    {
+      id: 'resources',
+      title: 'Resources',
+      active: activeTab === 'resources',
+      onSelect: () => setActiveTab('resources'),
+    },
+  ];
```

Comment:
So much repetition in these tab definitions (same with the previous file). Maybe we can also have a `tab(id, title, extra)` helper to reduce repetition here and over there?